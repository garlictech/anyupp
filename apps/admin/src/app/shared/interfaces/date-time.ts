export interface IDayInterval {
  start: number;
  end: number;
};