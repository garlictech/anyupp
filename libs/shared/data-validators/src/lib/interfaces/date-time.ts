import * as Joi from 'joi';

export const dateStringSchema = Joi.string().pattern(
  /^([1-9][0-9]{3})-(0?[1-9]|1[012])-(0?[1-9]|[12][0-9]|3[01])$/,
  'YYYY-MM-DD format',
);
export const timeStringSchema = Joi.string().pattern(
  /^(?:\d|[01]\d|2[0-3]):[0-5]\d$/,
  'HH:MM format',
);
