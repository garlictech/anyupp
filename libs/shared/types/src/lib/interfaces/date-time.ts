export interface DayInterval {
  start: number;
  end: number;
}
