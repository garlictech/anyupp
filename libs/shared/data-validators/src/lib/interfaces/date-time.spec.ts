import { dateStringSchema, timeStringSchema } from './date-time';

describe('DateTime -', () => {
  describe('dateStringSchema validaton test', () => {
    it('should be INVALID', () => {
      expect(dateStringSchema.validate(12)).toMatchSnapshot();
      expect(dateStringSchema.validate('STRING')).toMatchSnapshot();
      expect(dateStringSchema.validate('0000-00-00')).toMatchSnapshot();
      expect(dateStringSchema.validate('1234-13-01')).toMatchSnapshot();
      expect(dateStringSchema.validate('1234-12-32')).toMatchSnapshot();
    });

    it('should be VALID', () => {
      expect(dateStringSchema.validate('2010-12-01')).toMatchSnapshot();
    });
  });

  describe('timeStringSchema validaton test', () => {
    it('should be INVALID', () => {
      expect(timeStringSchema.validate(12)).toMatchSnapshot();
      expect(timeStringSchema.validate('STRING')).toMatchSnapshot();
      expect(timeStringSchema.validate('24:01')).toMatchSnapshot();
      expect(timeStringSchema.validate('23:61')).toMatchSnapshot();
    });

    it('should be VALID', () => {
      expect(timeStringSchema.validate('00:00')).toMatchSnapshot();
      expect(timeStringSchema.validate('01:01')).toMatchSnapshot();
      expect(timeStringSchema.validate('23:59')).toMatchSnapshot();
    });
  });
});
