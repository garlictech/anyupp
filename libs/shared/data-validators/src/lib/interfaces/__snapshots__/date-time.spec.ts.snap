// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DateTime - dateStringSchema validaton test should be INVALID 1`] = `
Object {
  "error": [ValidationError: "value" must be a string],
  "value": 12,
}
`;

exports[`DateTime - dateStringSchema validaton test should be INVALID 2`] = `
Object {
  "error": [ValidationError: "value" with value "STRING" fails to match the YYYY-MM-DD format pattern],
  "value": "STRING",
}
`;

exports[`DateTime - dateStringSchema validaton test should be INVALID 3`] = `
Object {
  "error": [ValidationError: "value" with value "0000-00-00" fails to match the YYYY-MM-DD format pattern],
  "value": "0000-00-00",
}
`;

exports[`DateTime - dateStringSchema validaton test should be INVALID 4`] = `
Object {
  "error": [ValidationError: "value" with value "1234-13-01" fails to match the YYYY-MM-DD format pattern],
  "value": "1234-13-01",
}
`;

exports[`DateTime - dateStringSchema validaton test should be INVALID 5`] = `
Object {
  "error": [ValidationError: "value" with value "1234-12-32" fails to match the YYYY-MM-DD format pattern],
  "value": "1234-12-32",
}
`;

exports[`DateTime - dateStringSchema validaton test should be VALID 1`] = `
Object {
  "value": "2010-12-01",
}
`;

exports[`DateTime - timeStringSchema validaton test should be INVALID 1`] = `
Object {
  "error": [ValidationError: "value" must be a string],
  "value": 12,
}
`;

exports[`DateTime - timeStringSchema validaton test should be INVALID 2`] = `
Object {
  "error": [ValidationError: "value" with value "STRING" fails to match the HH:MM format pattern],
  "value": "STRING",
}
`;

exports[`DateTime - timeStringSchema validaton test should be INVALID 3`] = `
Object {
  "error": [ValidationError: "value" with value "24:01" fails to match the HH:MM format pattern],
  "value": "24:01",
}
`;

exports[`DateTime - timeStringSchema validaton test should be INVALID 4`] = `
Object {
  "error": [ValidationError: "value" with value "23:61" fails to match the HH:MM format pattern],
  "value": "23:61",
}
`;

exports[`DateTime - timeStringSchema validaton test should be VALID 1`] = `
Object {
  "value": "00:00",
}
`;

exports[`DateTime - timeStringSchema validaton test should be VALID 2`] = `
Object {
  "value": "01:01",
}
`;

exports[`DateTime - timeStringSchema validaton test should be VALID 3`] = `
Object {
  "value": "23:59",
}
`;
